guides/v2.3/graphql/develop/resolvers.md

```diff
@@ -42,7 +15,185 @@ $info | [`Magento\Framework\GraphQl\Schema\Type\ResolveInfo`]({{ site.mage2blobu
+
+[`\Magento\RelatedProductGraphQl\Model\Resolver\Batch\AbstractLinkedProducts`]({{ site.mage2bloburl }}/{{ page.guide_version }}/app/code/Magento/RelatedProductGraphQl/Model/Resolver/Batch/AbstractLikedProducts.php) contains an example of batch resolver implementation.
+
+### BatchServiceContractResolverInterface
+
+Requests for this interface to resolve are gathered into batches in the same way as for `BatchResolverInterface`, except that the actual resolving is delegated to a batch service contract. The job of the `BatchServiceContractResolverInterface` resolver is to convert GraphQL requests into DTOs acceptable by the service contract, and then convert results returned by the contract into a GraphQL response.
+
+Consider the same example query:
+
+```graphql
+query ($filter: ProductAttributeFilterInput!) {
+  products (filter: $filter) {
+    items {
+      id
+      sku
+      related_products {
+        sku
+        related_products {
+          sku
+        }
+      }
+    }
+    total_count
+  }
+}
+```
+
+Here, we will delegate loading all related products to a service that accepts a list of root product IDs, and then returns individual lists for each.
+
+Pseudo-code for a GraphQL resolver delegating the work to a service contract may look like this:
+
+```php
+class RelatedProductsResolver implements BatchServiceContractResolverInterface
+{
+    ...
+
+    public function getServiceContract(): array
+    {
+        return [ProductLinksRetriever::class, 'getRelatedProducts'];
+    }
+
+    public function convertToServiceArgument(ResolveRequestInterface $request)
+    {
+        return new RootProductCriteria($request->getValue()['model']->getId());
+    }
+
+    public function convertFromServiceResult($result, ResolveRequestInterface $request)
+    {
+        return $result->getLinkedProducts();
+    }
+}
+```
+
+The `getServiceContract()` method points to the service contract to be used.
+
+The `convertToServiceArgument()` method converts GraphQL requests to a criteria item to be passed in a list as the argument to the contract. Remember that batch service contract methods must accept a single argument: a list (array) of criteria objects.
+
+The `convertFromServiceResult()` method converts one of the result items into a GraphQL response (a [`\Magento\Framework\GraphQl\Query\Resolver\Value`]({{ site.mage2bloburl }}/{{ page.guide_version }}/lib/internal/Magento/Framework/GraphQl/Query/Resolver/Value.php) instance or an array). Remember that batch service contracts must return result items in the same order as were the criteria items passed as the method's list argument. For example, if the first root product had the ID #555, then the first result item will contain the list of related products to product #555.
```

Comment:
I get lost reading the second sentence. Does this mean the same thing?

> Remember, batch service contracts must return result items in the same order as the criteria items that were passed to the method's list argument.

Reply:
yes, just an example